### DISCUSSION TITLE
rxn.RunReactants loss their valence and H's information?

### DISCUSSION THREAD

**Discussion by JustinDoIt** [rdkit/rdkit#5422] [Q&A]
Hi,

I need some help to figure-out what I'm doing wrong(!). I tried to run the reaction as following

```python
from rdkit import Chem
from rdkit.Chem.rdChemReactions import ReactionFromSmarts

m = Chem.MolFromSmarts('[#8&v2&H1:1]-[#6&v4&H0:2](=[#8&v2&H0:3])-[#6&v4&H2:4]-[#6&v4&H2:5]-[#6&v4&H2:6]-[#6&v4&H0:7](=[#8&v2&H0:8])-[#6&v4&H2:9]-[#6&v4&H0:10](=[#8&v2&H0:11])-[#16&v2&H0:12]-[#6&v4&H2:13]-[#6&v4&H2:14]-[#7&v3&H0:15]=[#6&v4&H0:16](-[#8&v2&H1:17])-[#6&v4&H2:18]')
rxn = ReactionFromSmarts(f'[C:1](=[O:2])-[S]-[C]-[C]-[N]>>[C:1](=[O:2])-[At]')
prod = rxn.RunReactants((m,))[0][0]
print(Chem.MolToSmarts(prod))
```

I expected the output will **remain the valence and H's information** as it was in `prod`, such

```
'[#6](=[#8])(-[At])-[#6&v4&H2:9]-[#6&v4&H0:7](-[#6&v4&H2:6]-[#6&v4&H2:5]-[#6&v4&H2:4]-[#6&v4&H0:2](-[#8&v2&H1:1])=[#8&v2&H0:3])=[#8&v2&H0:8]'
```
However, I got
```
'[#6](=[#8])(-[At])-[#6:9]-[#6:7](-[#6:6]-[#6:5]-[#6:4]-[#6:2](-[#8:1])=[#8:3])=[#8:8]'
```

Why does this happen and how should I solve this issue? Thanks for any help!

**Comment by bp-kelley** [comment]
You are confusing queries with molecules.

While queries (smarts) define the reaction, they are designed to run on and output molecules, not queries.  As such, query information is not propagated to the product which is good, considering that the machinery does query->molecule matching not query->query matching.

**Reply by JustinDoIt** [reply]
> You are confusing queries with molecules.
> 
> While queries (smarts) define the reaction, they are designed to run on and output molecules, not queries. As such, query information is not propagated to the product which is good, considering that the machinery does query->molecule matching not query->query matching.

@bp-kelley Hi, thanks for your response

The reason for this is to modify a reaction template, which is represented with SMARTS: "SMARTS>>SMARTS"

replace the CoA (In biosynthesis, those with 'S-C-C-N' substructure) with a metal atom

as in, input:
```
([#8&v2&H1:1]-[#6&v4&H0:2](=[#8&v2&H0:3])-[#6&v4&H2:4]-[#6&v4&H2:5]-[#6&v4&H2:6]-[#6&v4&H0:7](=[#8&v2&H0:8])-[#6&v4&H2:9]-[#6&v4&H0:10](=[#8&v2&H0:11])-[#16&v2&H0:12]-[#6&v4&H2:13]-[#6&v4&H2:14]-[#7&v3&H0:15]=[#6&v4&H0:16](-[#8&v2&H1:17])-[#6&v4&H2:18])>>([#8&v2&H1:1]-[#6&v4&H0:2](=[#8&v2&H0:3])-[#6&v4&H2:4]-[#6&v4&H2:5]-[#6&v4&H2:6]-[#6&v4&H1:7](-[#8&v2&H1:8])-[#6&v4&H2:9]-[#6&v4&H0:10](=[#8&v2&H0:11])-[#16&v2&H0:12]-[#6&v4&H2:13]-[#6&v4&H2:14]-[#7&v3&H0:15]=[#6&v4&H0:16](-[#8&v2&H1:17])-[#6&v4&H2:18])
```
Plot it with `ReactionFromSmarts`:
![image](https://user-images.githubusercontent.com/25944216/178191348-431cff14-28a6-4def-85af-178f2d6dcac2.png)

What I want is:
![image](https://user-images.githubusercontent.com/25944216/178191460-a7642319-ee52-4ee9-9fac-43ebe5e26bae.png)

SO, I extract two parts in the reaction template. AND using query language to manipulate query language...

Are any other methods to do this?

**Reply by greglandrum** [reply]
@JustinDoIt at the moment it's not possible to do this with the RDKit. The reaction code is already pretty complex and trying to also support working with (or changing) query features of atoms would be difficult and error prone.
You can edit the molecule directly to achieve the same effect though. Here's a demo of how that could work:
https://gist.github.com/greglandrum/c51eec765f868303816ff211fd738ba2

Note that since your query pattern doesn't match all the atoms on the "other side" of the S atom, you end up with a molecule with multiple fragments. The chemical reaction logic would have done the same thing. You need to right some more code if you want that whole fragment to be removed.
